src/dir.cr

```diff
@@ -93,6 +96,24 @@ class Dir
     String.new(dir).tap { C.free(dir as Void*) }
   end
 
+  def self.chdir path
+    if C.chdir(path) != 0
+      raise Errno.new("Error while changing directory")
+    end
+  end
```

Comment:
I think it's just an implementation side-effect in Ruby, I see no actual value in it. You either successfully changed the directory or get an exception, there's no need to check whether you succeeded.